### DISCUSSION TITLE
how to initialize dropdown close <AutoCompleteItem>

### DISCUSSION THREAD

**Discussion by SergeyBiryukov34** [mantinedev/mantine#3433] [Q&A] [answered]
I'm using a <Autocomplete> component, with a custom "AutoCompleteItem" component that has a click event handler. How do I initialize the closing of the list when clicking on "AutoCompleteItem"

`    const AutoCompleteItem = forwardRef<HTMLDivElement, ISearch>(
        ({ name, country, region }: ISearch, ref) => (
            <div
                ref={ref}
                className={classes.AutoCompleteItem}
                onClick={() => {onSubmitName(name)}}
            >
                <Group noWrap>
                    <div>
                        <Text>{name}, {region}, {country}</Text>
                    </div>
                </Group>
            </div>
        )
    );`

`        <Autocomplete
            value={value}
            onChange={setValue}
            placeholder="Enter location name"
            itemComponent={AutoCompleteItem}
            data={data}
            rightSection={isLoading || isFetching ? <Loader/> : null}
            error={isError ? <p>Something went wrong...</p> : null}
            nothingFound={'Nothing Found'}
        />`

**Comment by Sajarin-M** [comment, marked as answer]
Use `onMouseDown` instead of `onClick` and call the received `onMouseDown` handler after doing your stuff
```js
const AutoCompleteItem = forwardRef<HTMLDivElement, ISearch>(
  ({ name, country, region, onMouseDown, ...others }: ISearch, ref) => (
    <div
      ref={ref}
      onMouseDown={(e) => {
        // do your stuff
        onMouseDown(e);
      }}
      {...others}
    >
      {/*  content */}
    </div>
  ),
);
```
Also make sure to spread the `others` props to the AutoCompleteItem component to ensure that any additional props passed to the component will be passed down

**Reply by SergeyBiryukov34** [reply]
Does not work setValue("")

**Reply by SergeyBiryukov34** [reply]
And the click event fires the second time or does not fire at all

**Reply by Sajarin-M** [reply]
Could you please provide more information, context, or a codesandbox demonstrating the issue you are facing?

**Reply by SergeyBiryukov34** [reply]
![Снимок экрана 2023-01-30 100447](https://user-images.githubusercontent.com/59689428/215410065-726bf6e2-5379-459e-8a1e-f5212e6a6dac.png)

**Reply by SergeyBiryukov34** [reply]
When pressed 'setValue' does not work

**Reply by Sajarin-M** [reply]
This approach will not work because we are calling `onMouseDown(e)` after `setValue("")`, which sets the value of the selected item. To resolve this, you may swap the order of these two calls.
```js
onMouseDown(e);
setValue("");
```
Alternatively, consider using the `onItemSubmit` prop on the Autocomplete component, as it will simplify your code and allow the AutoCompleteItem to focus solely on design.
```js
<Autocomplete
  onItemSubmit={(item)=> {
    // do something
  }}
/>
